JS-Algorithms/challenges.js

```diff
@@ -70,7 +87,28 @@ The last 3 digits for the sum of powers from 1 to 10 is "317"
 ***** */
 
 const ownPower = (number, lastDigits) => {
-  // YOUR CODE HERE...
+  const modBase = Math.pow(10, lastDigits + 1);
+
+  const serie = Array.from({ length: number }, (_, idx) => idx + 1);
+
+  const poweredSerie = serie.map((num) => {    
+    let poweredNumber = num;
+
+    for (let i = 1; i < num; i++) {
+      poweredNumber *= num;
+      poweredNumber %= modBase;
+    }
+
+    return poweredNumber;
+  });
+
+  const sumOfPowers = poweredSerie
+    .reduce((acc, currNum) => (acc + currNum) % modBase, 0)
+    .toString();
+
+  const boundedSumOfPowers = sumOfPowers.slice(sumOfPowers.length - lastDigits);
+
+  return boundedSumOfPowers;
```

Comment:
Good solution, I like the different approach maybe for future references bear in mind that some of the functions can be chained

```
const modBase = Math.pow(10, lastDigits + 1);
const sumOfPowers = Array.from({ length: number }, (_, idx) => {
    let poweredNumber = 1;
    for (let i = 0; i < idx + 1; i++) {
      poweredNumber = (poweredNumber * (idx + 1)) % modBase;
    }
    return poweredNumber;
  })
    .reduce((acc, currentNumber) => (acc + currentNumber) % modBase, 0)
    .toString();
    
   return sumOfPowers.slice(-lastDigits);
```

Reply:
At first, I thought about sending it like that, but I wasn't sure how readable it would be. I'll keep it in mind for next time.